### PR TITLE
feat(core): turn on batch mode by default

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: e2e/js/src/js-inlining.test.ts [hunk 1]
```diff
@@ -57,7 +57,11 @@ describe('inlining', () => {
 
       // 1. external is set to all
       execSync(`rm -rf dist`);
-      runCLI(`build ${parent} --external=all`);
+      runCLI(`build ${parent} --external=all`, {
+        env: {
+          NX_BATCH_MODE: 'false',
```
Comment (AgentEnder) [inline e2e/js/src/js-inlining.test.ts:62]: Why is batch mode disabled for these tests?

File: packages/nx/src/tasks-runner/run-command.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ export async function runCommand(
 }
 
 function setEnvVarsBasedOnArgs(nxArgs: NxArgs, loadDotEnvFiles: boolean) {
-  if (nxArgs.outputStyle == 'stream' || process.env.NX_BATCH_MODE === 'true') {
```
Comment (AgentEnder) [inline packages/nx/src/tasks-runner/run-command.ts:197]: It seems like this functionality is lost now - previously batch mode meant streamed logs, now it doesn't. I wonder if there's a good way to check if any task will use batch mode

File: packages/js/src/executors/tsc/lib/batch/build-task-info-per-tsconfig-map.ts [hunk 3]
```diff
@@ -105,6 +109,17 @@ function createTaskInfo(
...
 
+  const tsLibDependency = getHelperDependency(
+    HelperDependency.tsc,
+    taskOptions.tsConfig,
+    buildableProjectNodeDependencies,
+    context.projectGraph
+  );
+
+  if (tsLibDependency) {
+    buildableProjectNodeDependencies.push(tsLibDependency);
```
Comment (FrozenPandaz) [inline packages/js/src/executors/tsc/lib/batch/build-task-info-per-tsconfig-map.ts:120]: We shouldn't need to do this anymore after @jaysoo removes that tsc option.

File: packages/nx/src/tasks-runner/run-command.ts (hunk 2 above)
Comment (FrozenPandaz) [reply packages/nx/src/tasks-runner/run-command.ts:197]: We might be able to only set stream when we actually run the batch.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
In a separate PR, breakout jest into 2 executors.

**Pull request closed without merging**